fix(passport): ignore static state and nonce passed to Strategy() (#556)
panva/node-openid-client: docs/README.md

---

> The strategy automatically generates `state` and `nonce` parameters when required. To provide one for a flow where it is optional (for example the `nonce` for the Authorization Code Flow), it can be passed in the optional `options` argument to `passport.authenticate()`:
> 
> ```js
> app.post('/auth/oidc', function(req, res, next) {
>   passport.authenticate('oidc', { nonce: crypto.randomBytes(16).toString('base64url') })(req, res, next);
> });
> ```

Reviewer: ```suggestion
Authorization request parameters can be passed in the optional `options` argument to `passport.authenticate()`, this way you can set them per transaction:

```js
app.get('/protected-route', function(req, res, next) {
  if (shouldReConsent(req)) {
    passport.authenticate('oidc', { prompt: 'consent' })(req, res, next);
  }
});
```
```
